src/storage/RepresentationConvertingStore.ts

```diff
@@ -0,0 +1,43 @@
+import { Conditions } from './Conditions';
+import { matchingMediaType } from '../util/Util';
+import { PassthroughStore } from './PassthroughStore';
+import { Representation } from '../ldp/representation/Representation';
+import { RepresentationConverter } from './conversion/RepresentationConverter';
+import { RepresentationPreferences } from '../ldp/representation/RepresentationPreferences';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { ResourceStore } from './ResourceStore';
+
+/**
+ * Store that overrides the `getRepresentation` function.
+ * Tries to convert the {@link Representation} it got from the source store
+ * so it matches one of the given type preferences.
+ */
+export class RepresentationConvertingStore extends PassthroughStore {
+  private readonly converter: RepresentationConverter;
+
+  public constructor(source: ResourceStore, converter: RepresentationConverter) {
+    super(source);
+    this.converter = converter;
+  }
+
+  public async getRepresentation(identifier: ResourceIdentifier, preferences: RepresentationPreferences,
+    conditions?: Conditions): Promise<Representation> {
+    const representation = await super.getRepresentation(identifier, preferences, conditions);
+    if (this.matchesPreferences(representation, preferences)) {
+      return representation;
+    }
+    return this.converter.handleSafe({ representation, preferences });
+  }
+
+  private matchesPreferences(representation: Representation, preferences: RepresentationPreferences): boolean {
```

Comment:
So at the moment, we're doing "lowest effort" conversion, which is fine for now. However, `RepresentationConvertingStore` is the _one_ store that will try hard to satisfy user agent preferences (as opposed to the others, which only do minimum effort). Perhaps add a comment about this.

Reply:
Not sure I completely understand. What exactly should be added to the comment on line 12?

Reply:
Well, all other stores will do minimum effort to satisfy user preferences, if at all.

`RepresentationConvertingStore`, in the future, will try to make sure people get what they want.

Right now, when the user agent says `text/turtle;q=1,application/ld+json;q=0.1` and the original source gives JSON-LD, you will just happily server them JSON-LD because they accept it. But at some point, we should really try to give them Turtle if that is what they want.